src/event-publisher/index.js

```diff
@@ -1,234 +1,102 @@
-import WebSocket from 'ws'
-import { HttpsProxyAgent } from 'https-proxy-agent'
-import { createMessage } from '../messaging/create-message.js'
+import { getEventPublisher } from '../messaging/fcp-messaging-service.js'
 
 export const SEND_SESSION_EVENT = 'send-session-event'
 export const APPLICATION_STATUS_EVENT = 'application-status-event'
 
-const eventQueueConfig = config.get('azure.eventQueue')
-
 export const raiseApplicationStatusEvent = async (event) => {
-  const eventBatch = [
-    {
-      name: SEND_SESSION_EVENT,
-      properties: {
-        id: `${event.application.id}`,
-        sbi: `${event.application.organisation.sbi}`,
-        cph: 'n/a',
-        checkpoint: process.env.APPINSIGHTS_CLOUDROLE || 'placeholder', //TODO can we delete this?
-        status: 'success',
-        action: {
-          type: `application:status-updated:${event.application.status}`,
-          message: event.message,
-          data: {
-            reference: event.application.reference,
-            status: event.application.status
-          },
-          raisedBy: event.raisedBy,
-          raisedOn: event.raisedOn.toISOString()
-        }
-      }
-    }
-  ]
-  // TODO: status history lives with application/claim now so we don;t need to emit anything elsewhere
-  // Tidy this up when ensuring that is saved
-  // if (config.storeHistoryInDb.enabled) {
-  //   await createStatusHistory({
-  //     reference: event.application.reference,
-  //     statusId: event.application.statusId,
-  //     note: event.note,
-  //     createdAt: event.raisedOn.toISOString(),
-  //     createdBy: event.raisedBy
-  //   })
-  // } else {
-  //   eventBatch.unshift({
-  //     name: APPLICATION_STATUS_EVENT,
-  //     properties: {
-  //       id: `${event.application.id}`,
-  //       sbi: `${event.application.data.organisation.sbi}`,
-  //       cph: 'n/a',
-  //       checkpoint: process.env.APPINSIGHTS_CLOUDROLE,
-  //       status: 'success',
-  //       action: {
-  //         type: 'status-updated',
-  //         message: event.message,
-  //         data: {
-  //           reference: event.application.reference,
-  //           statusId: event.application.statusId,
-  //           note: event.note
-  //         },
-  //         raisedBy: event.raisedBy,
-  //         raisedOn: event.raisedOn.toISOString()
-  //       }
-  //     }
-  //   })
-  // }
-
-  const messages = eventBatch.map((message) => {
-    message.properties.action.timestamp = new Date().toISOString()
-    return createMessage(
-      message,
-      message.properties.action.type,
-      message.properties.checkpoint
-    )
+  await getEventPublisher().publishEvent({
+    name: SEND_SESSION_EVENT,
+    id: `${event.application.id}`,
+    sbi: `${event.application.organisation.sbi}`,
+    cph: 'n/a',
+    checkpoint: process.env.APPINSIGHTS_CLOUDROLE,
+    status: 'success',
+    type: `application:status-updated:${event.application.status}`,
```

Comment:
We are not going to be able to suddenly start populating the status Id in the MI report with the name of the status, we will need to still convert this into the ID somewhere sadly. Perhaps we should do that in the event function itself so we can keep that stuff self-contained, away from the CDP world